File: atcoder-wiki/backend/model/pages.go
package model

import "time"

type Page struct {
	ID        int       `json:"id"`
	Slug      string    `json:"slug"`
	Title     string    `json:"title"`
	ContentMD string    `json:"content_md"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: atcoder-wiki/frontend/src/api/pages.ts
import type { Page } from "../types/page"


const BASE_URL = "http://localhost:8080"

export async function createPage(data: {
  title: string
  content_md: string
  tags: string[]
}): Promise<Page> {
  const res = await fetch(`${BASE_URL}/api/pages`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(data),
  })

  if (!res.ok) {
    throw new Error("failed to create page")
  }

  return res.json()
}

export async function fetchPages(): Promise<Page[]> {
    const res = await fetch(`${BASE_URL}/api/pages`)
  
    if (!res.ok) {
      throw new Error("failed to fetch pages")
    }
  
    return res.json()
  }
